hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/BlockRecordManagerImpl.java

```diff
@@ -295,6 +301,18 @@ public Instant firstConsTimeOfLastBlock() {
         return BlockRecordInfoUtils.firstConsTimeOfLastBlock(lastBlockInfo);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public Instant consTimeOfLastHandledTxn() {
+        final var lastHandledTxn = lastBlockInfo.consTimeOfLastHandledTxn();
+        return lastHandledTxn != null
```

Comment:
Do we set this to null ? I thought in migration we set as `Instant.EPOCH`

Reply:
Yes you're right, we should also check for `Instant.EPOCH`. I'll fix this

Reply:
Ohhh, this is only a defensive check. `consTimeOfLastHandledTxn` shouldn't ever be null, but if it is, we'll still return `Instant.EPOCH`.

Reply:
Agreed, but if it is null, its a bug we should fix. Just confuses about the possibility of being null.

Reply:
I could add a comment if that's preferable. `null` is one of those things that is always a possibility, whether there's code handling it or not 🤷‍♂️ I'd hate for it to fail in production because of an edge case. Or I can take it out if we really want to do that.